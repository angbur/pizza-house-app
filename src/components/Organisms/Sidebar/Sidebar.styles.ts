import { getInitialTheme } from './../../Theme/initialTheme';
import styled from 'styled-components';

export const SidebarNav = styled.div`
  width: 114px;
  height: 100vh;
  position: sticky;
  border-right: 1px solid ${getInitialTheme().palette.light};
`;
