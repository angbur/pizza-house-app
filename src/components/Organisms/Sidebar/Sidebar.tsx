import ListIconItem from 'components/Molecules/ListIconItem/ListIconItem';
import { ThemeContext } from 'components/Theme/ThemeContext';
import { useContext } from 'react';
import styled from 'styled-components';

const SidebarNav = styled.div`
  width: 114px;
  position: sticky;
  height: 100%;
  border-right: 1px solid ${(props) => props.theme.palette.light};
  & > ul {
    list-style: none;
  }
`;

const Sidebar = () => {
  const theme = useContext(ThemeContext);
  return (
    <nav>
      <SidebarNav className={'sidebar'} theme={theme}>
        <ul>
          <ListIconItem icon={{ name: 'menu', size: 32 }} label={'menu'} linkTo={'/'} />
          <ListIconItem
            icon={{ name: 'pizza-circle', size: 32 }}
            label={'make pizza'}
            linkTo={'make-pizza'}
          />
          <ListIconItem icon={{ name: 'order', size: 26 }} label={'order'} linkTo={'order'} />
          <ListIconItem
            icon={{ name: 'list-icon', size: 20 }}
            label={'history'}
            linkTo={'history'}
          />
          <ListIconItem icon={{ name: 'email', size: 20 }} label={'contact'} linkTo={'contact'} />
        </ul>
      </SidebarNav>
    </nav>
  );
};

export default Sidebar;
