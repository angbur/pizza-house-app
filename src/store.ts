import { configureStore } from '@reduxjs/toolkit';
import orderReducer from 'components/Pages/Order/orderSlice';
import { pizzaApi } from 'services/pizza';

export const store = configureStore({
  reducer: {
    order: orderReducer,
    [pizzaApi.reducerPath]: pizzaApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(pizzaApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
