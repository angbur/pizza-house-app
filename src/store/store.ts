import { configureStore, Store } from '@reduxjs/toolkit';
import orderReducer from 'components/Pages/Order/orderSlice';
import dialogReducer from 'components/Organisms/Dialog/dialogSlice';
import { pizzaApi } from 'services/pizza';
import orderMiddleware from './middleware/orderMiddleware';

export const store: Store = configureStore({
  reducer: {
    order: orderReducer,
    dialog: dialogReducer,
    [pizzaApi.reducerPath]: pizzaApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(pizzaApi.middleware, orderMiddleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
