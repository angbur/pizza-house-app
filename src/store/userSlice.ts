import { createSlice } from '@reduxjs/toolkit';
import { RootState } from './store';
import { toast } from 'react-toastify';
import { userApi } from 'services/user';

type User = {
  loggedIn: boolean;
  status: string | undefined;
  error: string | undefined;
  userId: string | undefined;
  login: string | undefined;
  token: string | undefined;
};

const initialState: User = {
  loggedIn: false,
  status: 'idle',
  error: undefined,
  userId: undefined,
  login: undefined,
  token: undefined,
};

const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    logout: (state) => {
      state.loggedIn = false;
      state.userId = undefined;
      state.login = undefined;
      state.token = undefined;
      toast.success('Logged out!');
    },
  },
  extraReducers: (builder) => {
    builder.addMatcher(userApi.endpoints.login.matchFulfilled, (state, { payload }) => {
      state.loggedIn = true;
      state.token = payload.token;
      state.userId = payload.userId;
      state.login = payload.login;
    });
  },
});

export const { logout } = userSlice.actions;

export const selectName = (state: RootState) => state.user.user;

export const isLoggedIn = (state: RootState) => state.user.loggedIn;

export const selectResponseStatus = (state: RootState) => state.user.status;

export const selectResponseError = (state: RootState) => state.user.error;

export const selectLoggedInUser = (state: RootState) => state.user.user;

export default userSlice.reducer;
