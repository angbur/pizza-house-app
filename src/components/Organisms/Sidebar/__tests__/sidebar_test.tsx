import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import { store } from 'store/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Sidebar from '../Sidebar';

const Wrapper = () => {
  return (
    <ThemeProvider>
      <BrowserRouter>
        <Provider store={store}>
          <Sidebar />
        </Provider>
      </BrowserRouter>
    </ThemeProvider>
  );
};

describe('Sidebar', () => {
  it('should renders sidebar correctly', () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchInlineSnapshot(`
      <nav>
        <div
          className="sc-gswNZR fXfFKY sidebar"
        >
          <ul>
            <li>
              <a
                aria-current="page"
                className="active"
                href="/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="sc-bcXHqe WDvVZ icon-item"
                >
                  <img
                    height={32}
                    src={
                      Object {
                        "ReactComponent": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                        "__esModule": true,
                        "default": "menu-active.svg",
                      }
                    }
                  />
                  <h5
                    className="item-title"
                    style={
                      Object {
                        "color": "none",
                        "fontFamily": "Source Sans Pro",
                        "fontSize": "1rem",
                        "fontWeight": "600",
                        "letterSpacing": "1px",
                        "lineHeight": "1.4",
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    menu
                  </h5>
                </div>
              </a>
            </li>
            <li>
              <a
                className=""
                href="/make-pizza"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="sc-bcXHqe gIzvfo icon-item"
                >
                  <img
                    height={32}
                    src={
                      Object {
                        "ReactComponent": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                        "__esModule": true,
                        "default": "pizza-circle.svg",
                      }
                    }
                  />
                  <h5
                    className="item-title"
                    style={
                      Object {
                        "color": "none",
                        "fontFamily": "Source Sans Pro",
                        "fontSize": "1rem",
                        "fontWeight": "600",
                        "letterSpacing": "1px",
                        "lineHeight": "1.4",
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    make pizza
                  </h5>
                </div>
              </a>
            </li>
            <li>
              <a
                className=""
                href="/order"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="sc-bcXHqe gIzvfo icon-item"
                >
                  <img
                    height={26}
                    src={
                      Object {
                        "ReactComponent": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                        "__esModule": true,
                        "default": "order.svg",
                      }
                    }
                  />
                  <h5
                    className="item-title"
                    style={
                      Object {
                        "color": "none",
                        "fontFamily": "Source Sans Pro",
                        "fontSize": "1rem",
                        "fontWeight": "600",
                        "letterSpacing": "1px",
                        "lineHeight": "1.4",
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    order
                  </h5>
                </div>
              </a>
            </li>
            <li>
              <a
                className=""
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="sc-bcXHqe gIzvfo icon-item"
                >
                  <img
                    height={20}
                    src={
                      Object {
                        "ReactComponent": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                        "__esModule": true,
                        "default": "email.svg",
                      }
                    }
                  />
                  <h5
                    className="item-title"
                    style={
                      Object {
                        "color": "none",
                        "fontFamily": "Source Sans Pro",
                        "fontSize": "1rem",
                        "fontWeight": "600",
                        "letterSpacing": "1px",
                        "lineHeight": "1.4",
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    contact
                  </h5>
                </div>
              </a>
            </li>
          </ul>
        </div>
      </nav>
    `);
  });
});
