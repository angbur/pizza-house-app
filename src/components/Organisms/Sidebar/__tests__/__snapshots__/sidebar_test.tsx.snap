// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should renders sidebar correctly 1`] = `
<nav>
  <div
    className="sc-gswNZR fXfFKY sidebar"
  >
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-bcXHqe WDvVZ icon-item"
          >
            <img
              height={32}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "menu-active.svg",
                }
              }
            />
            <h5
              className="item-title"
              style={
                Object {
                  "color": "none",
                  "fontFamily": "Source Sans Pro",
                  "fontSize": "1rem",
                  "fontWeight": "600",
                  "letterSpacing": "1px",
                  "lineHeight": "1.4",
                  "textTransform": "uppercase",
                }
              }
            >
              menu
            </h5>
          </div>
        </a>
      </li>
      <li>
        <a
          className=""
          href="/make-pizza"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-bcXHqe gIzvfo icon-item"
          >
            <img
              height={32}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "pizza-circle.svg",
                }
              }
            />
            <h5
              className="item-title"
              style={
                Object {
                  "color": "none",
                  "fontFamily": "Source Sans Pro",
                  "fontSize": "1rem",
                  "fontWeight": "600",
                  "letterSpacing": "1px",
                  "lineHeight": "1.4",
                  "textTransform": "uppercase",
                }
              }
            >
              make pizza
            </h5>
          </div>
        </a>
      </li>
      <li>
        <a
          className=""
          href="/order"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-bcXHqe gIzvfo icon-item"
          >
            <img
              height={26}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "order.svg",
                }
              }
            />
            <h5
              className="item-title"
              style={
                Object {
                  "color": "none",
                  "fontFamily": "Source Sans Pro",
                  "fontSize": "1rem",
                  "fontWeight": "600",
                  "letterSpacing": "1px",
                  "lineHeight": "1.4",
                  "textTransform": "uppercase",
                }
              }
            >
              order
            </h5>
          </div>
        </a>
      </li>
      <li>
        <a
          className=""
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-bcXHqe gIzvfo icon-item"
          >
            <img
              height={20}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "email.svg",
                }
              }
            />
            <h5
              className="item-title"
              style={
                Object {
                  "color": "none",
                  "fontFamily": "Source Sans Pro",
                  "fontSize": "1rem",
                  "fontWeight": "600",
                  "letterSpacing": "1px",
                  "lineHeight": "1.4",
                  "textTransform": "uppercase",
                }
              }
            >
              contact
            </h5>
          </div>
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
